packages/leap/lib/src/leap_map.dart

```diff
@@ -42,16 +45,41 @@ class LeapMap extends PositionComponent with HasGameRef<LeapGame> {
   }
 
   /// Convenience method for accessing Tiled layers in the [tiledMap].
-  T? getTileLayer<T extends Layer>(String name) {
-    return tiledMap.tileMap.getLayer<T>(name);
+  T getTileLayer<T extends Layer>(String name) {
+    // First try to find the layer with the exact name.
+    final layer = tiledMap.tileMap.getLayer<T>(name);
+    if (layer != null) {
+      return layer;
+    }
+
+    // Then try to find the layer with the name in all lower case.
+    final nameLowerCased = name.toLowerCase();
+    if (tiledMap.tileMap.getLayer<T>(nameLowerCased) != null) {
+      return tiledMap.tileMap.getLayer<T>(nameLowerCased)!;
+    }
+
+    // Then try to find the layer with the name in all upper case.
+    final nameUpperCased = name.toUpperCase();
+    if (tiledMap.tileMap.getLayer<T>(nameUpperCased) != null) {
+      return tiledMap.tileMap.getLayer<T>(nameUpperCased)!;
+    }
+
+    // Finally, throw an error if the layer can't be found.
+    Error.throwWithStackTrace(
+      Exception(
+        'Layer $name not found, check the Tiled map for the correct name.',
+      ),
+      StackTrace.current,
+    );
```

Comment:
I don't think this is a good idea because each time you call `getLayer` it loops over all the layers inside the map, so we don't want to call `getLayer` more than once. Honestly I wouldn't bother support case insensitive names, but if you want to I would add it to the `flame_tiled` library itself so it can do it in one loop.